crd-generator/api-v2/src/main/java/io/fabric8/crdv2/generator/AbstractCustomResourceHandler.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crdv2.generator;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+import java.util.stream.Stream;
+
+/**
+ * This class encapsulates the common behavior between different CRD generation logic. The
+ * intent is that each CRD spec version is implemented as a sub-class of this one.
+ */
+public abstract class AbstractCustomResourceHandler {
```

Comment:
Is there any reason why this is a class instead of an interface?

It's my understanding that we can have a  `interface CustomResourceHandler` and then an implementation `V1CustomResourceHandler` or `CustomResourceHandlerV1`, right?

Reply:
> Is there any reason why this is a class instead of an interface?

Not really. It used to hold common state and have a common method for dealing with printer columns. My intent was to add back the method for printer columns, but didn't bother yet.

> It's my understanding that we can have a interface CustomResourceHandler and then an implementation V1CustomResourceHandler or CustomResourceHandlerV1, right?

Correct, it seemed fine to leave in the abstraction layer, used here and for AbstractJsonSchema, but it's not really needed at the moment.